MITADS-Speech/corpora_importer.py

```diff
@@ -0,0 +1,282 @@
+                .encode('latin1')         # 1:1 mapping back to bytes
+                .decode(encoding))        # Decode original encoding
+    except:
+        return (s.encode('utf-8')  ## cp1252     
+                .decode('unicode-escape') 
+                .encode('utf-8')  
+                .decode(encoding))   
+
+def get_counter():
+    return Counter({'all': 0, 'failed': 0, 'invalid_label': 0, 'too_short': 0, 'too_long': 0, 'imported_time': 0, 'total_time': 0})
+
+
+class Corpus:
+    def __init__(self,utterences:dict,audios:list,datasets_sizes = [0.8,0.1,0.1],make_wav_resample=False): 
+
+        ##utterences:dict --> key: audio_file_full_path  value:utterance 
+        ## audios is a list of all audio_file , full absolute path
+        ## datasets_sizes: dimension of train-test-dev datasets (in different csv)
+        self.utterences = utterences
+        self.audios = audios
+        self.datasets_sizes = datasets_sizes
+        self.make_wav_resample = make_wav_resample
+
+
+class ArchiveImporter:
+    def __init__(self,corpus_name,archive_url,data_dir=None,csv_append_mode=False):
+        self.corpus_name=corpus_name
+        self.archive_url=archive_url
+        ##Make archive_name from archive_filename
+        archive_filename = self.archive_url.rsplit('/', 1)[-1]    
+        self.archive_name = archive_filename.rsplit('.', 1)[0]
+        # Making path absolute root data or prefered from param data_dir
+        self.dataset_path = os.path.abspath(self.corpus_name) if data_dir==None else  os.path.join(data_dir, self.corpus_name)
+        
+        self.origin_data_path = os.path.join(self.dataset_path, "origin") if data_dir==None else  data_dir
+        
+        self.dataset_output_path = os.path.abspath(self.corpus_name)
+        self.csv_append_mode = csv_append_mode
+
+    def run(self):
+        self._download_and_preprocess_data()
+
+    def _download_and_preprocess_data(self):
+
+        if not path.exists(self.dataset_output_path):
+            print('No path "%s" - creating ...' % self.dataset_output_path)
+            makedirs(self.dataset_output_path)
+
+        archive_filename = self.archive_url.rsplit('/', 1)[-1]
+        # Conditionally download data
+        archive_path = maybe_download(archive_filename, self.origin_data_path, self.archive_url)
+        # Conditionally extract common voice data
+        self._maybe_extract(self.origin_data_path, self.archive_name, archive_path)
+       
+        ##get corpus:  audio_file_names + transcriptions
+        print('Filter audio file and parse transcript...')
+        corpus = self.get_corpus()
+
+         # Conditionally convert CSV files and mp3/wav data to DeepSpeech CSVs and wav
+        self._maybe_convert_sets(corpus)      
+
+    def _maybe_extract(self,target_dir, extracted_data, archive_path):
+        # If target_dir/extracted_data does not exist, extract archive in target_dir
+        extracted_path = os.path.join(target_dir, extracted_data)
```

Comment:
```suggestion
```suggestion
    def _maybe_extract(self,target_dir, extract_dir, archive_path):
        # If target_dir/extract_dir does not exist, extract archive in target_dir
        extracted_path = os.path.join(target_dir, extract_dir)
```
```